plugin/core/configurations.py

```diff
@@ -73,6 +81,34 @@ def disable_config(self, config_name: str, only_for_session: bool = False) -> No
             disable_in_project(self._window, config_name)
         self.update(config_name)
 
+    def record_crash(self, config_name: str, exit_code: int, exception: Optional[Exception]) -> bool:
+        """
+        Signal that a session has crashed.
+
+        Returns True if the session should be restarted automatically.
+        """
+        now = datetime.now()
+        if config_name not in self._crashes:
+            self._crashes[config_name] = deque([now], maxlen=RETRY_MAX_COUNT)
+        else:
+            self._crashes[config_name].append(now)
+        timeout = now - RETRY_COUNT_TIMEDELTA
+        crash_count = len([crash for crash in self._crashes[config_name] if crash > timeout])
+        msg = "".join((
+            "session for config {} crashed ",
+            "({} / {} times in the last {} seconds), ",
+            "exit code {}, exception: {}",
+        )).format(
+            config_name,
+            crash_count,
+            RETRY_MAX_COUNT,
+            RETRY_COUNT_TIMEDELTA.total_seconds(),
+            exit_code,
+            exception
+        )
+        printf(msg)
```

Comment:
So many lines... Can we just do this?
Note that the message is slightly altered. I think that "session for config" is a bit too low-level. Just printing the name of the config should be enough.

```suggestion
        printf("{} crashed ({} / {} times in the last {} seconds), exit code {}, exception: {}".format(
            config_name, crash_count, RETRY_MAX_COUNT, RETRY_COUNT_TIMEDELTA.total_seconds(), exit_code, exception))
```

Reply:
Sure.